src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -3047,6 +3066,9 @@ subroutine gsi_fv3ncdf_read_ens_parallel_over_ens(filenamein,fv3filenamegin, &
           if( present(dbz) )then            ! phyvars: dbz
             dbz = hwork
           end if
+          if( present(fed) )then            ! phyvars: fed 
```

Comment:
@hongli-wang  could you run a test using both dbz and fed?   By doing so, the above issues would be apparent .

Reply:
@TingLei-NOAA 
Thanks for all the great comments!

The current code base is not very user friendly to add capability that might be conflict with dbz DA. We can work together to make the code well organized. 

I will find time to address your concerns. 

Regarding this comment, the case you mentioned has been tested, I didn't see the issue though it does have a change dbz will be overwritten with FED when both dbz and fed are .true. . I will double check.  

If  dbz and fed is not defined separatetively in varname_files(see below). It will definitely have issue and need to add ivar=1 .or.2 to help identify correct variable when both are save with varname_files(2) . 


if( present(dbz) )then            ! phyvars: dbz
          allocate(varname_files(1))
          varname_files = (/'ref_f3d'/)
       end if
       if( present(fed) )then            ! phyvars: fed

Reply:
@TingLei-NOAA 
If both dbz and fed are true, this will not cause issues because, dbz and fed are read separately: 
             if(if_model_dbz) then
               call gsi_fv3ncdf_read_ens_parallel_over_ens(fv3_filenameginput%phyvars,fv3_filenameginput,dbz=g_dbz,iope=iope)
             end if
             if(if_model_fed) then
               call gsi_fv3ncdf_read_ens_parallel_over_ens(fv3_filenameginput%phyvars,fv3_filenameginput,fed=g_fed,iope=iope)
             end if

Also,  varname_files are defined separatetely in varname_files(see below). 
if( present(dbz) )then ! phyvars: dbz
allocate(varname_files(1))
varname_files = (/'ref_f3d'/)
end if
if( present(fed) )then ! phyvars: fed
varname_files = (/'fed'/)
end if
So the current code wont have any issues when both dbz and fed are true.

Reply:
@TingLei-NOAA 
The code is written this way is to reduce interaciton with the dbz code, leave the dbz code unchanged as much as possible. 

I can try to merge the code to one call, and define both dbz and fed in varname_files. But if this won't improve the code efficiency, I will not consider to do the change.  Do you think call gsi_fv3ncdf_read_ens_parallel_over_ens twice will slow the code? 

Thanks, 
Hongli

Reply:
@hongli-wang  let us assume both l_model_dbz and l_model_fed =.true.  Then, 
```
          if( present(dbz) )then            ! phyvars: dbz
            dbz = hwork
          end if
          if( present(fed) )then            ! phyvars: fed 
                      fed = hwork
          end if
```
will make fed=dbz=hwork,  do you agree with this?

Reply:
@TingLei-NOAA 
Sure, I will combine two calls to one call. Thanks again for your suggestion.

Reply:
> This part should be moved into the 2981 do block . It seems in those parts, the case when both dbz and fed exist is not considered.

@TingLei-NOAA 
Double check. do you mean inside  the do loop
     do ivar = 1, size(varname_files)
....
     end do 

It is in the above do loop.